include/writer/kernel.tpp

```diff
@@ -38,17 +38,17 @@ void specfem::writer::kernel<OutputLibrary>::write() {
   Kokkos::parallel_reduce(
       "specfem::writer::kernel", specfem::kokkos::HostRange(0, nspec),
       [=](const int ispec, int &n_elastic_isotropic, int &n_elastic_anisotropic, int &n_acoustic) {
-        if (kernels.h_element_types(ispec) ==
+        if (kernels.h_medium_tags(ispec) ==
```

Comment:
Can you call `kernels.get_element_on_host` here?

Reply:
do you want to create a function count_elements_on_host()?

Reply:
You wont need that for this case. Since on line 73 you call `get_element_on_host` you can replace this Kokkos loop with those functions. And then use `int n_elastic_anisotropic = element_indices.extent(0)`